src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,1001 @@ static ColliderCylinderInit D_809D5C40 = {
+                                // Set the Arwing to fly in a circle around the player.
+                                targetCircleX = Math_SinS(player->actor.shape.rot.y) * 400.0f;
+                                targetCircleZ = Math_CosS(player->actor.shape.rot.y) * 400.0f;
+                                this->targetPosition.x =
+                                    Rand_CenteredFloat(700.0f) + (player->actor.world.pos.x + targetCircleX);
+                                this->targetPosition.y = Rand_ZeroFloat(200.0f) + player->actor.world.pos.y + 150.0f;
+                                this->targetPosition.z =
+                                    Rand_CenteredFloat(700.0f) + (player->actor.world.pos.z + targetCircleZ);
+                            } else {
+                                // Set the Arwing to fly to a random position.
+                                this->targetPosition.x = Rand_CenteredFloat(700.0f);
+                                this->targetPosition.y = Rand_ZeroFloat(200.0f) + 150.0f;
+                                this->targetPosition.z = Rand_CenteredFloat(700.0f);
+                            }
+                        }
+
+                        this->targetDirection.z = 0.0f;
+                        this->targetDirection.y = 0.0f;
+                        this->targetDirection.x = 0.0f;
+                    }
+
+                    rotationScale = 10;
+                    xRotationTarget = 2048;
+                    loseTargetLockDistance = 100.0f;
+                    if (this->state == CLEAR_TAG_STATE_TARGET_LOCKED) {
+                        // Set the Arwing to fly towards the player.
+                        this->targetPosition.x = player->actor.world.pos.x;
+                        this->targetPosition.y = player->actor.world.pos.y + 40.0f;
+                        this->targetPosition.z = player->actor.world.pos.z;
+                        rotationScale = 7;
+                        xRotationTarget = 4096;
+                        loseTargetLockDistance = 150.0f;
+                    } else if (this->state == CLEAR_TAG_STATE_DEMO) {
+                        // Move the Arwing for the intro cutscene.
+
+                        // Do a Barrel Roll!
+                        this->roll += 0.5f;
+                        if (this->roll > M_PI * 2) {
+                            this->roll -= M_PI * 2;
+                        }
+
+                        // Set the Arwing to fly to a hardcoded position.
+                        this->targetPosition.x = 0.0f;
+                        this->targetPosition.y = 300.0f;
+                        this->targetPosition.z = 0.0f;
+                        loseTargetLockDistance = 100.0f;
+                    }
+
+                    // If the Arwing is not in cutscene mode update the roll to zero.
+                    if (this->state != CLEAR_TAG_STATE_DEMO) {
+                        Math_ApproachZeroF(&this->roll, 0.1f, 0.2f);
```

Comment:
same here, the code pretty clearly says this

Reply:
I think it would be better to explain why this is important.